### PR TITLE
Datanode periodical for certificate renewals

### PR DESCRIPTION
**Pull request by todvora**
Add 30m peridocal in datanodes for triggering certificate signing requests. Remove duplicated and unsed code from the graylog-server side of the renewal handling.
/nocl

## Motivation and Context
CSRs should be triggered from datanodes as datanodes know best when their certificates expire and have all the means to create and trigger CSRs.

## How Has This Been Tested?
Added unit test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: data-node/src/main/java/org/graylog/datanode/bootstrap/preflight/DataNodeCertRenewalPeriodical.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                    switch (renewalPolicy.mode()) {
+                        case AUTOMATIC -> automaticRenewal();
+                        case MANUAL -> manualRenewal();
+                    }
+                });
+
+    }
+
+    private void manualRenewal() {
+        LOG.debug("Manual renewal, ignoring on the datanode side for now");
```
Comment (moesterheld) [inline data-node/src/main/java/org/graylog/datanode/bootstrap/preflight/DataNodeCertRenewalPeriodical.java:73]: maybe make that a WARN to give a clear indication to the user?

File: data-node/src/main/java/org/graylog/datanode/opensearch/statemachine/OpensearchStateMachine.java [hunk 2]
```diff
@@ -108,7 +108,8 @@ public static OpensearchStateMachine createNew(OpensearchProcess process, Set<St
                 .permit(OpensearchEvent.HEALTH_CHECK_OK, OpensearchState.AVAILABLE)
                 .permit(OpensearchEvent.PROCESS_STOPPED, OpensearchState.TERMINATED)
                 .permit(OpensearchEvent.PROCESS_PREPARED, OpensearchState.PREPARED) //restart if reconfigured
-                .permit(OpensearchEvent.PROCESS_TERMINATED, OpensearchState.TERMINATED);
+                .permit(OpensearchEvent.PROCESS_TERMINATED, OpensearchState.TERMINATED)
+                .permit(OpensearchEvent.PROCESS_STARTED, OpensearchState.STARTING);
```
Comment (moesterheld) [inline data-node/src/main/java/org/graylog/datanode/opensearch/statemachine/OpensearchStateMachine.java:112]: why did you add the STARTED transition here? i don't see where it is triggered

**Review by moesterheld — COMMENTED** [review]
works and looks good.
one suggestion and one question, though.

File: data-node/src/main/java/org/graylog/datanode/opensearch/statemachine/OpensearchStateMachine.java (hunk 2 above)
Comment (todvora) [reply data-node/src/main/java/org/graylog/datanode/opensearch/statemachine/OpensearchStateMachine.java:112]: I encountered a situation where the datanode startup failed repeatedly due to invalid certificate. Then I started the datanode server, this generated an usable cert for the datanode. Datanode could consume the cert but failed to start, because transitions from terminated to starting were not allowed.

File: data-node/src/main/java/org/graylog/datanode/bootstrap/preflight/DataNodeCertRenewalPeriodical.java (hunk 1 above)
Comment (todvora) [reply data-node/src/main/java/org/graylog/datanode/bootstrap/preflight/DataNodeCertRenewalPeriodical.java:73]: The user gets the notification from a periodical in the server. So I think we shouldn't make users worry about it. This is more like a TODO for us, if we decide that we want to support cluster-wide notifications from the datanode.

**Review by moesterheld — APPROVED** [review]
lgtm
